pkg/server/gqllogging.go

```diff
@@ -0,0 +1,42 @@
+package server
+
+import (
+	"context"
+	"time"
+
+	"github.com/99designs/gqlgen/graphql"
+	"github.com/99designs/gqlgen/graphql/handler/extension"
+	"go.uber.org/zap"
+)
+
+// NewGQLResponseMiddleware returns a handler with a request based logger
+func NewGQLResponseMiddleware(logger *zap.Logger) graphql.ResponseMiddleware {
+	return func(ctx context.Context, next graphql.ResponseHandler) *graphql.Response {
+
```

Comment:
The logger that you can access via the context will already be decorated with other request-related values (e.g. a requestID, etc), giving you richer association capabilities.

```suggestion
func NewGQLResponseMiddleware() graphql.ResponseMiddleware {
	return func(ctx context.Context, next graphql.ResponseHandler) *graphql.Response {
		logger := appcontext.ZLogger(ctx)
```

Reply:
Thanks for catching this!